### PR TITLE
New security response mapper mechanism for MP

### PR DESCRIPTION
**Pull request by spericas**
New security response mapper mechanism that enables MP applications to provide custom responses to users when a security problem (e.g., authentication error) is encountered. New integration test.

Signed-off-by: Santiago Pericasgeertsen <santiago.pericasgeertsen@oracle.com>

### CODE REVIEW

File: security/integration/jersey/src/main/java/io/helidon/security/integration/jersey/SecurityFilterCommon.java [hunk 1]
```diff
@@ -344,7 +345,11 @@ protected void abortRequest(SecurityFilter.FilterContext context,
             updateHeaders(responseHeaders, responseBuilder);
         }
 
-        if (featureConfig.isDebug()) {
+        // Run security response mappers if available, or revert to old logic for compatibility
+        ServiceLoader<SecurityResponseMapper> loader = ServiceLoader.load(SecurityResponseMapper.class);
```
Comment (tomas-langer) [inline security/integration/jersey/src/main/java/io/helidon/security/integration/jersey/SecurityFilterCommon.java:349]: The service loader should be used only  once, and in a static section.
Also please use `HelidonServiceLoader` as it supports prioritization and some basic control through system properties.

File: tests/integration/security/security-response-mapper/pom.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    <description>
+        Integration test for security response mappers
+    </description>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.helidon.microprofile.bundles</groupId>
+            <artifactId>helidon-microprofile</artifactId>
+        </dependency>
+        <dependency>
```
Comment (tomas-langer) [inline tests/integration/security/security-response-mapper/pom.xml:40]: Why is there a dependency on EL implementation? That should not be required by this module at all.

File: tests/integration/security/security-response-mapper/src/test/java/io/helidon/tests/integration/security/mapper/RestrictedResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import io.helidon.microprofile.server.ServerCdiExtension;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+class RestrictedResourceTest {
```
Comment (tomas-langer) [inline tests/integration/security/security-response-mapper/src/test/java/io/helidon/tests/integration/security/mapper/RestrictedResourceTest.java:35]: If you use `@HelidonTest`, you do not need beans.xml (you can add required beans using `@AddBean`), you also do not need to manually start/stop the container and create a web target (`WebTarget` can be injected into constructor, with the URI of the server prepared)

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: security/integration/jersey/src/main/java/io/helidon/security/integration/jersey/SecurityFilterCommon.java (hunk 1 above)
Comment (spericas) [reply security/integration/jersey/src/main/java/io/helidon/security/integration/jersey/SecurityFilterCommon.java:349]: Yes, good point

File: tests/integration/security/security-response-mapper/pom.xml (hunk 2 above)
Comment (spericas) [reply tests/integration/security/security-response-mapper/pom.xml:40]: Copy/paste from the other test. Will fix.

File: tests/integration/security/security-response-mapper/src/test/java/io/helidon/tests/integration/security/mapper/RestrictedResourceTest.java (hunk 3 above)
Comment (spericas) [reply tests/integration/security/security-response-mapper/src/test/java/io/helidon/tests/integration/security/mapper/RestrictedResourceTest.java:35]: Copy/paste from the other test. Will fix.

**Review by tomas-langer — APPROVED** [review]
LGTM

**Pull request merged**